### PR TITLE
Improve assert messaging (sortingcomponents)

### PR DESCRIPTION
**Pull request by zm711**
Just editing more assert messaging in sorting components this time.

### CODE REVIEW

File: src/spikeinterface/sortingcomponents/peak_selection.py [hunk 1]
```diff
@@ -252,8 +252,19 @@ def select_peak_indices(peaks, method, seed, **method_kwargs):
                 selected_indices = [rng.permutation(my_selection)[: params["n_peaks"]]]
 
     else:
-        raise NotImplementedError(f"No method {method} for peaks selection")
+        raise NotImplementedError(
+            f"The mthod {method} does not exist for peaks selection." f" possible methods are {_possible_methods}"
```
Comment (zm711) [inline src/spikeinterface/sortingcomponents/peak_selection.py:256]:
```suggestion
            f"The 'method' {method} does not exist for peaks selection." f" possible methods are {_possible_methods}"
```

File: src/spikeinterface/sortingcomponents/motion_estimation.py [hunk 2]
```diff
@@ -1452,7 +1461,7 @@ def scipy_conv1d(input, weights, padding="valid"):
         input = np.pad(input, [*[(0, 0)] * (input.ndim - 1), (padding, padding)])
         length_out = length - (kernel_size - 1) + 2 * padding
     else:
-        raise ValueError(f"Unknown padding {padding}")
+        raise ValueError(f"Unknown padding {padding}, padding must be 'same', 'valid' or an integer")
```
Comment (zm711) [inline src/spikeinterface/sortingcomponents/motion_estimation.py:1464]:
```suggestion
        raise ValueError(f"Unknown 'padding' value of {padding}, 'padding' must be 'same', 'valid' or an integer")
```

File: src/spikeinterface/sortingcomponents/motion_estimation.py [hunk 3]
```diff
@@ -1371,7 +1380,7 @@ def normxcorr1d(
         conv1d = scipy_conv1d
         npx = np
     else:
-        raise ValueError(f"Unknown conv_engine {conv_engine}")
+        raise ValueError(f"Unknown conv_engine {conv_engine}. conv_engine must be 'torch' or 'numpy'")
```
Comment (zm711) [inline src/spikeinterface/sortingcomponents/motion_estimation.py:1383]:
```suggestion
        raise ValueError(f"Unknown conv_engine {conv_engine}. 'conv_engine' must be 'torch' or 'numpy'")
```

File: src/spikeinterface/sortingcomponents/motion_estimation.py [hunk 4]
```diff
@@ -824,7 +824,7 @@ def compute_pairwise_displacement(
     from scipy import sparse
     from scipy import linalg
 
-    assert conv_engine in ("torch", "numpy")
+    assert conv_engine in ("torch", "numpy"), f"conv_engine must be 'torch' or 'numpy'"
```
Comment (zm711) [inline src/spikeinterface/sortingcomponents/motion_estimation.py:827]:
```suggestion
    assert conv_engine in ("torch", "numpy"), f"'conv_engine' must be 'torch' or 'numpy'"
```

File: src/spikeinterface/sortingcomponents/matching/naive.py [hunk 5]
```diff
@@ -44,7 +44,7 @@ def initialize_and_check_kwargs(cls, recording, kwargs):
         d = cls.default_params.copy()
         d.update(kwargs)
 
-        assert d["waveform_extractor"] is not None
+        assert d["waveform_extractor"] is not None, "waveform_extractor must be supplied"
```
Comment (zm711) [inline src/spikeinterface/sortingcomponents/matching/naive.py:47]:
```suggestion
        assert d["waveform_extractor"] is not None, "'waveform_extractor' must be supplied"
```

File: src/spikeinterface/sortingcomponents/matching/main.py [hunk 6]
```diff
@@ -35,7 +35,7 @@ def find_spikes_from_templates(recording, method="naive", method_kwargs={}, extr
     """
     from .method_list import matching_methods
 
-    assert method in matching_methods, "The method %s is not a valid one" % method
+    assert method in matching_methods, f"The method {method} is not valid. Use a method from {matching_methods}"
```
Comment (zm711) [inline src/spikeinterface/sortingcomponents/matching/main.py:38]:
```suggestion
    assert method in matching_methods, f"The 'method' {method} is not valid. Use a method from {matching_methods}"
```

File: src/spikeinterface/sortingcomponents/features_from_peaks.py [hunk 7]
```diff
@@ -61,6 +65,9 @@ def compute_features_from_peaks(
         extract_dense_waveforms,
     ]
     for feature_name in feature_list:
+        assert (
+            feature_name in _features_class.keys()
+        ), f"Feature {feature_name} is not possible. Possible features are {_features_class.keys()}"
```
Comment (zm711) [inline src/spikeinterface/sortingcomponents/features_from_peaks.py:70]:
```suggestion
        ), f"Feature {feature_name} in 'feature_list' is not possible. Possible features are {list(_features_class.keys())}"
```

File: src/spikeinterface/sortingcomponents/clustering/circus.py [hunk 8]
```diff
@@ -64,13 +64,13 @@ def _check_params(cls, recording, peaks, params):
         elif params["waveform_mode"] == "shared_memory":
             assert tmp_folder is None, "tmp_folder must be None for shared_memory"
         else:
-            raise ValueError("shared_memory")
+            raise ValueError("waveform_mode must be 'memmap' or 'shared_memory'")
```
Comment (zm711) [inline src/spikeinterface/sortingcomponents/clustering/circus.py:67]:
```suggestion
            raise ValueError("'waveform_mode' must be 'memmap' or 'shared_memory'")
```

**Review by samuelgarcia — APPROVED** [review]

**Comment by samuelgarcia** [comment]
Merci beaucoup.

**Pull request merged**